src/crypto/crypto_kdf.cc

```diff
@@ -0,0 +1,131 @@
+#include "crypto/crypto_kdf.h"
+#include "ncrypto.h"
+#include "v8.h"
+
+#include <openssl/core_names.h>     /* OSSL_KDF_* */
+
+namespace node {
+namespace crypto {
+
+v8::MaybeLocal<v8::Value> KDFTraits::EncodeOutput(
+    Environment* env,
+    const KDFConfig& params,
+    ByteSource* out) {
+  return out->ToArrayBuffer(env);
+}
+
+v8::Maybe<void> KDFTraits::AdditionalConfig(
+    CryptoJobMode mode,
+    const v8::FunctionCallbackInfo<v8::Value>& args,
+    unsigned int offset,
+    KDFConfig* params) {
+  Environment* env = Environment::GetCurrent(args);
+
+  params->mode = mode;
+
+  CHECK(args[offset]->IsString());  // Algorithm
+  CHECK(args[offset + 1]->IsObject());  // Params
+  CHECK(args[offset + 2]->IsUint32());  // Length
+
+  Utf8Value algorithm(env->isolate(), args[offset]);
+  params->kdfCtx = ncrypto::KdfCtxPointer::FromName(algorithm.ToStringView());
+
+  // if (!pass.CheckSizeInt32()) [[unlikely]] {
+  //   THROW_ERR_OUT_OF_RANGE(env, "pass is too large");
+  //   return v8::Nothing<void>();
+  // }
+  // if (!salt.CheckSizeInt32()) [[unlikely]] {
+  //   THROW_ERR_OUT_OF_RANGE(env, "salt is too large");
+  //   return v8::Nothing<void>();
+  // }
```

Comment:
I can still do these checks by checking if the param name is "pass" or "salt" but I think it's pointless

Reply:
Similarly, these changes will need to be ifdef guarded.